<?xml version="1.0"?>
<launch>

<node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen"      name="laser_filter">
        <remap from="/scan" to="/roy/scan" />
        <remap from="/scan_filtered" to="/roy/scan_filtered" />
        <remap from="/tf" to="/roy/tf" />
        <remap from="/lidar/front/scan" to="/roy/lidar/front/scan" />
    </node>
    
<node pkg="autonomy_leg_detection" type="leg_detection" name="leg_detection" output="screen">
        <param name="show_viz" value="false"/>
        <param name="show_marker" value="true"/>
        <param name="laserFeature/arc_min_aperture" value="1.57"/>
        <param name="laserFeature/arc_max_aperture" value="2.375"/>
        <param name="laserFeature/arc_std_max" value="0.05"/>
        <param name="laserFeature/segmentation_threshold" value="300.0"/>
        <param name="laserFeature/line_min_distance" value="170.0"/>
        <param name="laserFeature/line_error_threshold" value="5.0"/>
        <param name="laserFeature/max_leg_diameter" value="200.0"/>
        <param name="laserFeature/min_leg_diameter" value="50.0"/>
        <param name="featureLegTracker/leg_clean_ticks" value="2.0"/>
        <param name="featureLegTracker/person_clean_ticks" value="2.0"/>
        <param name="featureLegTracker/leg_update_radius" value="500.0"/>
        <param name="featureLegTracker/person_radius" value="500.0"/>
        <param name="marker_scale" value="0.5"/>
    <remap from="/leg/marker" to="/roy/leg/marker" />
    <remap from="/legs" to="/roy/legs" />
    <remap from="/scan_filtered" to="/roy/scan_filtered" />
    <remap from="/scan" to="/roy/scan" />

    </node>

    


  <node name="Assertive_Behaviour" pkg="Assertive_Behaviour" type="assertive_behaviour" output="screen">
   <remap from="/legs" to="/roy/legs" />
   <remap from="/scan" to="/roy/scan" />
   <remap from="/scan_filtered" to="/roy/scan_filtered" />
   <remap from="/sonar" to="/roy/sonar" />
   <remap from="/sound_player" to="/roy/sound_player" />
   <remap from="/cmd_vel" to="/roy/cmd_vel" />
   <remap from="/gripper_control" to="/roy/gripper_control" />
   <remap from="/leds/set_led" to="/roy/leds/set_led" />
   <remap from="/sound_player" to="/roy/sound_player" />
   <remap from="/global_poses" to="/vicon/roy/roy" />
  </node>
</launch>
